kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/MetricsReportSuite.scala

```diff
@@ -0,0 +1,379 @@
+   * @param expectNonEmptyTimestampToVersionResolutionDuration whether we expect
+   *                                                           timestampToVersionResolutionDuration
+   *                                                           to be non-empty (should be true
+   *                                                           for any time-travel by timestamp
+   *                                                           queries)
+   * @param expectNonZeroLoadProtocolAndMetadataDuration whether we expect
+   *                                                     loadProtocolAndMetadataDuration to be
+   *                                                     non-zero (should be true except when an
+   *                                                     exception is thrown before log replay)
+   */
+  def checkSnapshotReport(
+    f: (Table, Engine) => Snapshot,
+    path: String,
+    expectException: Boolean,
+    expectedVersion: Optional[Long],
+    expectedProvidedTimestamp: Optional[Long],
+    expectNonEmptyTimestampToVersionResolutionDuration: Boolean,
+    expectNonZeroLoadProtocolAndMetadataDuration: Boolean
+  ): Unit = {
+
+    val (snapshotReport, duration, exception) = getSnapshotReport(f, path, expectException)
+
+    // Verify contents
+    assert(snapshotReport.tablePath == resolvePath(path))
+    assert(snapshotReport.operationType == "Snapshot")
+    exception match {
+      case Some(e) =>
+        assert(snapshotReport.exception().isPresent &&
+          Objects.equals(snapshotReport.exception().get(), e))
+      case None => assert(!snapshotReport.exception().isPresent)
+    }
+    assert(snapshotReport.reportUUID != null)
+    assert(Objects.equals(snapshotReport.version, expectedVersion),
+      s"Expected version $expectedVersion found ${snapshotReport.version}")
+    assert(Objects.equals(snapshotReport.providedTimestamp, expectedProvidedTimestamp))
+
+    // Since we cannot know the actual durations of these we sanity check that they are > 0 and
+    // less than the total operation duration whenever they are expected to be non-zero/non-empty
+    if (expectNonEmptyTimestampToVersionResolutionDuration) {
+      assert(snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.isPresent)
+      assert(snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.get > 0)
+      assert(snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.get <
+        duration)
+    } else {
+      assert(!snapshotReport.snapshotMetrics.timestampToVersionResolutionDuration.isPresent)
+    }
+    if (expectNonZeroLoadProtocolAndMetadataDuration) {
+      assert(snapshotReport.snapshotMetrics.loadInitialDeltaActionsDuration > 0)
+      assert(snapshotReport.snapshotMetrics.loadInitialDeltaActionsDuration < duration)
+    } else {
+      assert(snapshotReport.snapshotMetrics.loadInitialDeltaActionsDuration == 0)
+    }
+  }
+
+  test("SnapshotReport valid queries") {
```

Comment:
Could you also test: try and time travel by a bad version (e.g. version 55) or by a bad timestamp (e.g. the year 1985) but on a valid table?

e.g. you could rename this test to `SnapshotReport queries on valid table`

Reply:
So originally I did not add these tests because my main goal with the exception cases was to cover the 3 possible places snapshot construction could fail (outlined [here](https://docs.google.com/document/d/1W1NTmHg14qYq-QEIcDtFFSMy3B0GjrV82ll6BtX21fE/edit?tab=t.0#heading=h.wsxjbfksl0zh) in design doc) instead of trying to cover all possible errors/exceptions during snapshot construction.
- Error resolving timestamp --> version (only applicable for time-travel by timestamp)
- Error building the log segment
- Error constructing the snapshot (loading table protocol and metadata)

But I'm happy to add these two cases anyways. I can just add an additional test for them.